application/tests/test_bucketlist.py

```diff
@@ -0,0 +1,175 @@
+from application import create_app, db
+
+
+class BucketListTest(unittest.TestCase):
+
+    def setUp(self):
+        self.app = create_app(config_name="testing")
+        self.ctx = self.app.app_context()
+        self.client = self.app.test_client()
+        self.ctx.push()
+        db.drop_all()
+        db.create_all()
+        self.user ={
+                    "username":"angie", 
+                    "password":"angie"
+                    }
+        self.added_user = User(self.user["username"], 
+            self.user["password"])
+        db.session.add(self.added_user)
+        db.session.commit()
+        user = User.query.filter_by(username = 'angie').first()
+        self.bucketlist = {
+                        'name': 'Go to Canada.', 
+                        'description':"with an helicopter."
+                        }
+        self.token = user.generate_auth_token(user.id)
+
+    def test_create_bucketlist(self):
+        """ Tests bucketlist creation."""
+        resp = self.client.post('/bucketlists', 
+            data=json.dumps(self.bucketlist), 
+            content_type= "application/json", headers={
+            "Authorization":self.token
+            })
+        self.assertEqual(resp.status_code, 201)
+        result = json.loads(resp.data)
+        self.assertEqual(result['message'], 
+            "Bucketlist created successfully.")
+        self.assertEqual(result['name'], 
+            'Go to Canada.')
+        self.assertEqual(result['description'], 
+            "with an helicopter.")
+
+    def test_invalid_token(self):
+        """Test user providing an invalid token."""
+        resp = self.client.post('/bucketlists', 
+            data=json.dumps(self.bucketlist), 
+            content_type= "application/json", headers={
+            "Authorization":"self.token"
+            })
+        result = json.loads(resp.data)
+        self.assertEqual(resp.status_code, 401)
+        self.assertEqual(result['message'], 
+            "Invalid token. Please register or login.")
+    
+    def test_missing_data(self):
+        resp = self.client.post('/bucketlists', 
+            data=json.dumps({"name":"Go to Canada."}), 
+            content_type= "application/json", headers={
+            "Authorization":self.token
+            })
+
+        result = json.loads(resp.data)
+        self.assertEqual(result['error'],
+                         "missing data in request.")
+
+    def test_get_all_bucketlists(self):
+        """Tests getting the bucketlists."""
+        resp = self.client.post('/bucketlists', 
+            data=json.dumps(self.bucketlist), 
+            content_type= "application/json", headers={
+            "Authorization":self.token
+            })
+        self.assertEqual(resp.status_code, 201)
+        response = self.client.get('/bucketlists', headers={
+            "Authorization":self.token
+            })
+        result = json.loads(response.data)
+        self.assertEqual(response.status_code, 201)
```

Comment:
status code `201` is normally used for indicating resource creation. Hence for this, `200` would be more preferable considering its a get request